lib/url/URLParserPlugin.js

```diff
@@ -184,6 +184,86 @@ class URLParserPlugin {
 					relative
 				);
 				dep.loc = /** @type {DependencyLocation} */ (expr.loc);
+
+				// Handle prefetch/preload hints
+				if (importOptions) {
+					// Validate webpackPrefetch
+					if (
+						importOptions.webpackPrefetch !== undefined &&
+						importOptions.webpackPrefetch !== true
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPrefetch\` expected true, but received: ${importOptions.webpackPrefetch}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackPreload
+					if (
+						importOptions.webpackPreload !== undefined &&
+						importOptions.webpackPreload !== true
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPreload\` expected true, but received: ${importOptions.webpackPreload}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackFetchPriority
+					if (
+						importOptions.webpackFetchPriority !== undefined &&
+						(typeof importOptions.webpackFetchPriority !== "string" ||
+							!["high", "low", "auto"].includes(
+								importOptions.webpackFetchPriority
+							))
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackFetchPriority\` expected "low", "high" or "auto", but received: ${importOptions.webpackFetchPriority}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
+
+					// Validate webpackPreloadAs
+					if (
+						importOptions.webpackPreloadAs !== undefined &&
+						typeof importOptions.webpackPreloadAs !== "string"
+					) {
+						parser.state.module.addWarning(
+							new UnsupportedFeatureWarning(
+								`\`webpackPreloadAs\` expected a string, but received: ${importOptions.webpackPreloadAs}.`,
+								/** @type {DependencyLocation} */ (expr.loc)
+							)
+						);
+					}
```

Comment:
We'll also need `webpackPreloadType` - ideally we'll have the same attributes as `link` - https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/link

Reply:
added in e7f4a0d32ac69ea77fc7fbe8b5a51723bd70d8a9